### PR TITLE
#19970 — Enable white labeling

### PR DESCRIPTION
**Pull request by ivorpad**
This PR will add white-labeling options in Configuration

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Comment by ivorpad** [comment]
https://user-images.githubusercontent.com/52452/118734872-978d4380-b83f-11eb-809a-312312473ac9.mp4

**Review by alfredo-dotcms — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp [hunk 1]
```diff
@@ -60,289 +67,435 @@ final boolean hasAdminRole = user.isAdmin();
...
+            <input type="hidden" name="loginScreenLogoInput" id="loginScreenLogoInput" value="">	
+            <p style="margin-top: 1rem; color: grey;">This is the logo used for the login screen and communications (e.g. emails, etc)</p>
+            <br />
+            <div style="margin-top: 2rem;">
+               <label for="topNav_logo">
+               <input type="checkbox" name="topNav" id="topNav_logo" />
+               Override Navbar Logo
+               </label>
+               <br />
+               <p style="margin-top: 1rem; color: grey;">If you want to override DotCMS' Nav Bar logo check this option and upload a new image</p>
```
Comment (fmontes) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:370]: The text needs to be message keys.



```suggestion
               <p style="margin-top: 1rem; color: grey;">You can white-label your instance of DotCMS uploading a new logo.</p>
```

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp [hunk 2]
```diff
@@ -60,289 +67,435 @@ final boolean hasAdminRole = user.isAdmin();
...
+                  <img style="max-width: 200px; width: 100%;" border="1" hspace="0" src="<%= IMAGE_PATH %>/company_logo?img_id=<%= company.getCompanyId() %>&key=<%= ImageKey.get(company.getCompanyId()) %>" vspace="0" />
+               </div>
+            </div>
+            <dot-asset-drop-zone id="dot-asset-drop-zone-main" style="display: none;" drop-files-text="Drop Image" upload-file-text="Uploading Image..." display-indicator="true"></dot-asset-drop-zone>
+            <input type="hidden" name="loginScreenLogoInput" id="loginScreenLogoInput" value="">	
+            <p style="margin-top: 1rem; color: grey;">This is the logo used for the login screen and communications (e.g. emails, etc)</p>
+            <br />
+            <div style="margin-top: 2rem;">
+               <label for="topNav_logo">
+               <input type="checkbox" name="topNav" id="topNav_logo" />
```
Comment (fmontes) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:366]: This needs to be a dojo checkbox.

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp [hunk 3]
```diff
@@ -60,289 +67,435 @@ final boolean hasAdminRole = user.isAdmin();
...
+                  <h3 style="font-weight: normal; margin-bottom: 1rem;">Navbar Logo</h3>
+                  <div class="logo">
+                     <button class="logo__delete">&times;</button>
+                     <div class="logo__container" style="width: 200px; padding: 1rem; border-radius: 2px;">
+                        <img style="max-width: 200px; width: 100%;" border="1" hspace="0" src="<%= IMAGE_PATH %>/company_logo?img_id=<%= company.getCompanyId() %>&key=<%= ImageKey.get(company.getCompanyId()) %>" vspace="0" />
+                     </div>
+                  </div>
+                  <dot-asset-drop-zone style="display: none;" id="dot-asset-drop-zone-navbar" drop-files-text="Drop Image" upload-file-text="Uploading Image..." display-indicator="true"></dot-asset-drop-zone>
+                  <input type="hidden" name="topNavLogoInput" id="topNavLogoInput" value="">	
+                  <p style="margin-top: 1rem; color: grey;">Make sure the image is 258px wide and 48px tall</p>
```
Comment (fmontes) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:381]:
```suggestion
                  <p style="margin-top: 1rem; color: grey;">Your logo needs to be horizontal with at least 32:9 ratio</p>
```

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp [hunk 4]
```diff
@@ -1,53 +1,60 @@
-<%@ include file="/html/portlet/ext/cmsconfig/init.jsp" %>
```
Comment (fmontes) [inline dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:1]: https://user-images.githubusercontent.com/751424/118829612-7e09ed00-b87b-11eb-9287-ac75eff8d8a7.mp4

**Review by fmontes — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp (hunk 2 above)
Comment (ivorpad) [reply dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:366]: I couldn't make it work. Looking for feedback.

File: dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp [hunk 5]
```diff
@@ -60,289 +67,435 @@ final boolean hasAdminRole = user.isAdmin();
...
+         </div>
+         <div style="margin-left: 10rem;">
+            <h3 style="font-weight: normal; margin-bottom: 1rem;">Login Screen Logo</h3>
+            <div class="logo">
+               <button class="logo__delete">&times;</button>
+               <div class="logo__container" style="width: 200px; padding: 1rem; border-radius: 2px;">
+                  <img style="max-width: 200px; width: 100%;" border="1" hspace="0" src="<%= IMAGE_PATH %>/company_logo?img_id=<%= company.getCompanyId() %>&key=<%= ImageKey.get(company.getCompanyId()) %>" vspace="0" />
+               </div>
+            </div>
+            <dot-asset-drop-zone id="dot-asset-drop-zone-main" style="display: none;" drop-files-text="Drop Image" upload-file-text="Uploading Image..." display-indicator="true"></dot-asset-drop-zone>
```
[the remark this answers is not in the document]
Comment (ivorpad) [reply dotCMS/src/main/webapp/html/portlet/ext/cmsconfig/company.jsp:360]: What do you mean?

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Pull request merged**